zio-http/src/main/scala/zhttp/service/server/ServerRequestHandler.scala

```diff
@@ -33,17 +33,15 @@ final case class ServerRequestHandler[R](
     if (hh == null) {
       JWebSocketServerHandshakerFactory.sendUnsupportedVersionResponse(ctx.channel, ctx.channel().voidPromise())
     } else {
+      val pl = ctx.channel().pipeline()
+      pl.addLast(WEB_SOCKET_HANDLER, ServerSocketHandler(zExec, res.socket))
+      pl.remove(HTTP_REQUEST_HANDLER)
```

Comment:
We might not want to remove the Http Request Handler, because it's possible that the handshake fails and the client retries an HTTP request.

Reply:
We can remove the HTTP Request handler once the handshake succeeds 
```
     val pl = ctx.channel().pipeline()
      pl.addLast(WEB_SOCKET_HANDLER, ServerSocketHandler(zExec, res.socket))
      try {
        // handshake can throw
        hh.handshake(ctx.channel(), jReq).addListener { (future: JChannelFuture) =>
          if (!future.isSuccess) {
            ctx.fireExceptionCaught(future.cause)
            ()
          }else{
            pl.remove(HTTP_REQUEST_HANDLER)
            ()
          }

        }
      } catch {
        case _: JWebSocketHandshakeException =>
          JWebSocketServerHandshakerFactory.sendUnsupportedVersionResponse(ctx.channel, ctx.channel().voidPromise())
      }